core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -130,6 +136,19 @@ private DataMapStoreManager() {
     return allDataMaps;
   }
 
+  /**
+   * It gives all segment indexes, a map contains segment file name to SegmentIndexMeta.
+   *
+   * @return
+   */
+  public Map<String, SegmentIndexMeta> getAllSegmentIndexes(String tableId) {
+    if (segmentIndexes.get(tableId) == null) {
+      Map<String, SegmentIndexMeta> segmentIndexMetaMap = new ConcurrentHashMap<>();
+      segmentIndexes.putIfAbsent(tableId, segmentIndexMetaMap);
```

Comment:
Looks like each time, we are adding new map to `segmentIndexes`. where we actually add SegmentIndexMeta info to `segmentIndexes`?